core/log/logger.hpp

```diff
@@ -505,6 +519,20 @@ class EnableLogging : public Loggable {
         loggers_.push_back(logger);
     }
 
+    void remove_logger(const Logger *logger) override
+    {
+        auto idx = find_if(begin(loggers_), end(loggers_),
+                           [&logger](std::shared_ptr<const Logger> l) {
+                               return lend(l) == logger;
+                           });
+        if (idx != end(loggers_)) {
+            loggers_.erase(idx);
+        } else {
+            throw OutOfBoundsError(__FILE__, __LINE__, loggers_.size(),
+                                   loggers_.size());
```

Comment:
Do you really have to throw here? I don't know how others do it, but for example `std::remove` does not seem to throw in such cases.

https://en.cppreference.com/w/cpp/algorithm/remove

Reply:
With `std::remove`,  there is a special value returned if nothing is removed, meaning that if no one handles the return value, it just continues quietly. That is because not removing anything is not considered exceptional behavior, as it may just be used to make sure something is not contained in a range.

In our case, we could also return success/failure as a return value. However, removing a specific logger (meaning the exact same object) from something that does not already have it seems pretty exceptional to me, and if it happens, it's probably a bug in the program (I can't think of a common use case where you have a reference to a logger, and don't know at compile time if an object has that logger attached or not), so I would say that failing is a better default than continuing quietly.